File: dotnet/Multiplayer/Leaderboard/LeaderboardResponse.cs
using Michitai.Multiplayer.Errors;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Michitai.Multiplayer.Leaderboard
{
    /// <summary>
    /// Response containing the leaderboard entries with rankings.
    /// </summary>
    /// <typeparam name="T">The type to deserialize player data into.</typeparam>
    public class LeaderboardResponse<T> : ApiResponse<ELeaderboardError> where T : class, new()
    {
        /// <summary>
        /// List of leaderboard entries with player rankings.
        /// </summary>
        public List<LeaderboardPlayer<T>> Leaderboard { get; set; } = new();

        /// <summary>
        /// Total number of entries in the leaderboard.
        /// </summary>
        public int Total { get; set; }

        /// <summary>
        /// Array of field names used for sorting.
        /// </summary>
        public string[] Sort_by { get; set; } = Array.Empty<string>();

        /// <summary>
        /// The limit applied to the number of results.
        /// </summary>
        public int Limit { get; set; }
    }
}

File: java/src/main/java/com/michitai/multiplayer/leaderboard/LeaderboardResponse.java
package com.michitai.multiplayer.leaderboard;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.michitai.multiplayer.ApiResponse;

import java.util.List;

/**
 * Response containing the leaderboard entries with rankings.
 *
 * @param <T> The type to deserialize player data into.
 */
public class LeaderboardResponse<T> extends ApiResponse {
    @JsonProperty("leaderboard")
    private List<LeaderboardPlayer<T>> leaderboard;

    @JsonProperty("total")
    private int total;

    @JsonProperty("sort_by")
    private String[] sortBy;

    @JsonProperty("limit")
    private int limit;

    public List<LeaderboardPlayer<T>> getLeaderboard() {
        return leaderboard;
    }

    public void setLeaderboard(List<LeaderboardPlayer<T>> leaderboard) {
        this.leaderboard = leaderboard;
    }

    public int getTotal() {
        return total;
    }

    public void setTotal(int total) {
        this.total = total;
    }

    public String[] getSortBy() {
        return sortBy;
    }

    public void setSortBy(String[] sortBy) {
        this.sortBy = sortBy;
    }

    public int getLimit() {
        return limit;
    }

    public void setLimit(int limit) {
        this.limit = limit;
    }
}
